### DISCUSSION TITLE
Feature Request: Recognize device dashboard indicators as ROIs

### DISCUSSION THREAD

**Discussion by dm8tbr** [jomjol/AI-on-the-edge-device#2244] [Ideas]
### The Feature

I'm currently looking at using this to **read status panels of "white goods"** (washing machine, clothes dryer, dishwasher).
While I probably can work around limitations by essentially turning status LEDs and such into a "bit field of decimal digits 0/1" through custom models it would be nice to be able to expose multiple variables directly and without the need for external processing.

I'm happy to collaborate on this, but won't be able to bring you everything as a nice pull request.
Most likely I'll try to contribute some models with training data for digital displays, LEDs, indicators, and such.

For context a, completely blurry and unoptimized, picture of the washing machine dashboard I just took. It at least shows some lit/unlit LEDs and some non-digit LCD icons to give you an idea. I'll have to make a proper mount for the camera and focus it properly before I do actual work.

![Screenshot 2023-03-25 at 10-18-09 AI on the edge](https://user-images.githubusercontent.com/2073288/227706162-469d6f13-3e33-461c-bc2f-65923eb6ea65.png)
<img width="937" alt="image" src="https://user-images.githubusercontent.com/2073288/227711122-8af95083-40b8-456a-abf5-b03158a0c01e.png">

**Comment by caco3** [comment]
This is a special use case, I don't think we would add this to the normal firmware.

But your approach with custom models could work.

- For the LCD just use one of our Digit models.
- For the LEDs train your own analoque model. Basically it only would need to know 2 states. But if i understand correctly, the analoge models work with the pointer angle, which would not work for you. Maybe you would also have to use the digital model here as well and combine the LCD and the LEDs into a single model.

**Comment by dm8tbr** [comment]
Yeah, that's my current direction. As in most of my cases, I have only digital displays and LED/LCD indicators, so abusing the slot for the analog models will work.
What I was wondering though. Would it be possible to change how models are assigned? Overriding for a specific model per group of ROIs would increase flexibility a lot, not just for my use case.

**Reply by caco3** [reply]
> Would it be possible to change how models are assigned

Not sure what you mean by this.

The filter (which model is shown) is simply driven by dig/ana in the filename. Also, if a model does not have either pattern in its name, it gets shown on both ROI types. So renaming it should give you the flexibility.

**Comment by leifnel** [comment]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/9819287/503ea058-3c82-4e11-afac-117fcb7ae801)

It would be nice to be able to read the status indicators for water leak, no water pressure etc.
Mainly it should just detect something or nothing in the indicator area. The positions have to be guessed though, as it the "leak indicator" requires the water running for 24 hours continuously. Perhaps one should just go for "Hey, there's some alert, go look for yourself!" ;-)
Except the indicator for radio connection is not an error.

**Comment by sweti3** [comment]
@dm8tbr  - How did you do this? Trained a custom model? I have a topic - I want to read out my PV inverter, which luckily has a display. I want to read the dispaly (10000W is max). How did u do this? I would only need to somehow get these data into home assistant...
